Fatal Exception: java.lang.IllegalArgumentException

Version 1.0.5

1k+ crashes on prod. ;(

Fatal Exception: java.lang.IllegalArgumentException
Receiver not registered: h.c.a.a.b@903431
android.app.LoadedApk.forgetReceiverDispatcher (LoadedApk.java:789)
android.content.ContextWrapper.unregisterReceiver (ContextWrapper.java:576)
dev.fluttercommunity.plus.connectivity.ConnectivityBroadcastReceiver.onCancel (ConnectivityBroadcastReceiver.java)
dev.fluttercommunity.plus.connectivity.ConnectivityPlugin.teardownChannels (ConnectivityPlugin.java)
dev.fluttercommunity.plus.connectivity.ConnectivityPlugin.onDetachedFromEngine (ConnectivityPlugin.java)
io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove (FlutterEngineConnectionRegistry.java)
io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove (FlutterEngineConnectionRegistry.java)
io.flutter.embedding.engine.FlutterEngineConnectionRegistry.removeAll (FlutterEngineConnectionRegistry.java)
io.flutter.embedding.engine.FlutterEngineConnectionRegistry.destroy (FlutterEngineConnectionRegistry.java)
io.flutter.embedding.engine.FlutterEngine.destroy (FlutterEngine.java)
be.tramckrijte.workmanager.BackgroundWorker.stopEngine$lambda-0 (BackgroundWorker.java)
be.tramckrijte.workmanager.BackgroundWorker.lambda$LGzaYI9wpKnhzseAeQsBr_odg9A (BackgroundWorker.java)
be.tramckrijte.workmanager.-$$Lambda$BackgroundWorker$LGzaYI9wpKnhzseAeQsBr_odg9A.run (-.java)
android.os.Handler.handleCallback (Handler.java:815)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:679)

Same issue here with v1.0.5, it's crashing on multiple devices

<details>
  <summary>Error Log</summary>
  I/InputMethodManager(22338): com.abc.def called startInputInner(Do not restart input for non input field, ic=null) from android.view.inputmethod.InputMethodManager.restartInput(InputMethodManager.java:1188) <- io.flutter.embedding.android.FlutterView.detachFromFlutterEngine(FlutterView.java:980)
E/AndroidRuntime(22338): FATAL EXCEPTION: main
E/AndroidRuntime(22338): Process: com.abc.def, PID: 22338
E/AndroidRuntime(22338): java.lang.RuntimeException: Unable to destroy activity {com.abc.def/com.abc.def.MainActivity}: java.lang.IllegalArgumentException: Receiver not registered: dev.fluttercommunity.plus.connectivity.ConnectivityBroadcastReceiver@1541cdec
E/AndroidRuntime(22338): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4045)
E/AndroidRuntime(22338): 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4063)
E/AndroidRuntime(22338): 	at android.app.ActivityThread.access$1400(ActivityThread.java:144)
E/AndroidRuntime(22338): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1413)
E/AndroidRuntime(22338): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(22338): 	at android.os.Looper.loop(Looper.java:155)
E/AndroidRuntime(22338): 	at android.app.ActivityThread.main(ActivityThread.java:5696)
E/AndroidRuntime(22338): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(22338): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(22338): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1028)
E/AndroidRuntime(22338): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:823)
E/AndroidRuntime(22338): Caused by: java.lang.IllegalArgumentException: Receiver not registered: dev.fluttercommunity.plus.connectivity.ConnectivityBroadcastReceiver@1541cdec
E/AndroidRuntime(22338): 	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:828)
E/AndroidRuntime(22338): 	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1772)
E/AndroidRuntime(22338): 	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:510)
E/AndroidRuntime(22338): 	at dev.fluttercommunity.plus.connectivity.ConnectivityBroadcastReceiver.onCancel(ConnectivityBroadcastReceiver.java:70)
E/AndroidRuntime(22338): 	at dev.fluttercommunity.plus.connectivity.ConnectivityPlugin.teardownChannels(ConnectivityPlugin.java:58)
E/AndroidRuntime(22338): 	at dev.fluttercommunity.plus.connectivity.ConnectivityPlugin.onDetachedFromEngine(ConnectivityPlugin.java:36)
E/AndroidRuntime(22338): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(FlutterEngineConnectionRegistry.java:262)
E/AndroidRuntime(22338): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(FlutterEngineConnectionRegistry.java:270)
E/AndroidRuntime(22338): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.removeAll(FlutterEngineConnectionRegistry.java:278)
E/AndroidRuntime(22338): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.destroy(FlutterEngineConnectionRegistry.java:122)
E/AndroidRuntime(22338): 	at io.flutter.embedding.engine.FlutterEngine.destroy(FlutterEngine.java:436)
E/AndroidRuntime(22338): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach(FlutterActivityAndFragmentDelegate.java:576)
E/AndroidRuntime(22338): 	at io.flutter.embedding.android.FlutterActivity.release(FlutterActivity.java:595)
E/AndroidRuntime(22338): 	at io.flutter.embedding.android.FlutterActivity.onDestroy(FlutterActivity.java:616)
E/AndroidRuntime(22338): 	at android.app.Activity.performDestroy(Activity.java:6137)
E/AndroidRuntime(22338): 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1169)
E/AndroidRuntime(22338): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4025)
E/AndroidRuntime(22338): 	... 10 more
</details>

--- a/packages/connectivity_plus/connectivity_plus/CHANGELOG.md
+++ b/packages/connectivity_plus/connectivity_plus/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 1.0.6
+
+- Android: Fix unregisterReceiver call for API < N
+
 ## 1.0.5
 
 - Android: Fix memory leak from registered receiver

--- a/packages/connectivity_plus/connectivity_plus/android/src/main/java/dev/fluttercommunity/plus/connectivity/ConnectivityBroadcastReceiver.java
+++ b/packages/connectivity_plus/connectivity_plus/android/src/main/java/dev/fluttercommunity/plus/connectivity/ConnectivityBroadcastReceiver.java
@@ -67,7 +67,11 @@ public class ConnectivityBroadcastReceiver extends BroadcastReceiver
         networkCallback = null;
       }
     } else {
-      context.unregisterReceiver(this);
+      try {
+        context.unregisterReceiver(this);
+      } catch (Exception e) {
+        //listen never called, ignore the error
+      }
     }
   }
 

--- a/packages/connectivity_plus/connectivity_plus/pubspec.yaml
+++ b/packages/connectivity_plus/connectivity_plus/pubspec.yaml
@@ -1,6 +1,6 @@
 name: connectivity_plus
 description: Flutter plugin for discovering the state of the network (WiFi & mobile/cellular) connectivity on Android and iOS.
-version: 1.0.5
+version: 1.0.6
 homepage: https://plus.fluttercommunity.dev/
 repository: https://github.com/fluttercommunity/plus_plugins/tree/main/packages/
 

